src/symbolTable.ts

```diff
@@ -146,11 +151,18 @@ export class StringType extends Type {
       valueOf: this,
     };
   }
+
+  override getPropertyType(propertyName: string): Type | null {
+    let typeMap: TypeMap = {
+      length: new NumberType(),
+    };
+    return typeMap[propertyName] ?? null;
+  }
 }
 
 export class BooleanType extends Type {
   public toString() {
-    return "boolean";
+    return "BooleanType";
```

Comment:
Do we want the error messages to use this? I thought it was more readable the other way. I am not against the change, just curious about the thought process.

Reply:
For the simple types it doesn't matter too much; I was more concerned about array and union types where the toString() result could be unclear if there is lots of nesting going on. Then, I changed the rest of the values so that the capitalization and naming are uniform. (But we could easily drop the `Type` part and have `Boolean`, `Union`, etc. instead)